src/Storefront/Framework/Twig/TemplateDataExtension.php

```diff
@@ -109,7 +126,7 @@ protected function getThemeConfig(string $salesChannelId, ?string $themeId): arr
             return $themeConfig;
         }
 
-        $themePrefix = ThemeCompiler::getThemePrefix($salesChannelId, $themeId);
+        $themePrefix = ThemeCompiler::getThemePrefix($salesChannelId, (string) $themeName);
```

Comment:
$themeName is nullable. could this be a problem? maybe we should fallback to the $themeId?

Reply:
themeName should not be nullable

Reply:
It needs to be nullable to make an early return for the theme configuration. Sales channels without themes do not need all that. I have instead added a null check to the previously present check on `$themeId`, so that both need to be given.